subsys/bluetooth/host/conn.c

```diff
@@ -984,8 +1003,12 @@ void bt_conn_tx_processor(void)
 
 	LOG_DBG("processing conn %p", conn);
 
+	/* Sanity-check: do we have ownership? */
+	__ASSERT_NO_MSG(conn->ref > 0);
```

Comment:
Shouldn't this be > 1 to _somewhat_ assert ownership? 

If the conn exists without any pending TX, then the ref count is still at least 1. 

It may even be 10 or 20, as the reference counter can be incremented anywhere by anyone. 

Not sure if this check is really useful

Reply:
It was triggered without the fix in a bsim test (no PR for the test yet, I think). That means ref could be 1 here.

Reply:
But if we take a reference at connection creation (which we do), and take a reference when we add it to the TX list, and then see the value 1 here, doesn't that imply a missing ref or too many unrefs somewhere? 

If it is 1 here, and we unref it later in this function, then we may end with a ref count of 0. 

But I guess that may actually happened if we are in the disconnected state? In which case, this check should be moved to the `if (conn->state != BT_CONN_CONNECTED) {` clause, or be something like 
```c
__ASSERT_NO_MSG(conn->ref >= conn->state == BT_CONN_CONNECTED ? 2 : 1);
```

Reply:
Sure, if it's too much to request to fix, we can postpone it. 

It is (IMO) similar to a function that validates that a value is less than 10 implemented as
```
/* Checks if @p val is less than 10 */
bool foo(int val)
{
    return val < 20;
}
```

It's arguably better than not having the check a all (as it will properly check values like `100`), but it also does not check the value correctly if there are false positives which I believe there is.

Reply:
I just removed the assert. The compiler sees `{ }` in 99% cases anyways :)